homeassistant/components/yeelight/light.py

```diff
@@ -558,32 +564,30 @@ def device(self):
     def update(self):
         """Update light properties."""
         self._hs = self._get_hs_from_properties()
+        self._rgb = self._get_rgb_from_properties()
         if not self.device.is_color_flow_enabled:
             self._effect = None
 
     def _get_hs_from_properties(self):
-        rgb = self._get_property("rgb")
-        color_mode = self._get_property("color_mode")
-
-        if not rgb or not color_mode:
+        hue = self._get_property("hue")
+        sat = self._get_property("sat")
+        if hue is None or sat is None:
             return None
 
-        color_mode = int(color_mode)
-        if color_mode == 2:  # color temperature
-            temp_in_k = mired_to_kelvin(self.color_temp)
-            return color_util.color_temperature_to_hs(temp_in_k)
-        if color_mode == 3:  # hsv
-            hue = int(self._get_property("hue"))
-            sat = int(self._get_property("sat"))
+        return (int(hue), int(sat))
 
-            return (hue / 360 * 65536, sat / 100 * 255)
```

Comment:
This conversion seems wrong; Home Assistant expects hue in the range 0..360, not 0..65536 and a saturation in the range 0..100, not 0..255.
Since yeelight uses the same ranges, no conversion should be needed.